File: JUnitTestingLastAct/src/Singleton/Singleton.java
package Singleton;

public class Singleton {

    //create an object of SingleObject
    public static final Singleton Instance;

    public final String newMessage () {
        return "This is a Singleton";
    }
    //make the constructor private so that this class cannot be //instantiated
    // Exists only to defeat instantiation.
    private Singleton() {
    }

    static {
        Singleton newInstance = new Singleton();
       Instance = newInstance;
    }
}

File: KotlinTesting/src/Design Patterns/Creational/Singleton/Singleton.kt
package Singleton

class Singleton  //make the constructor private so that this class cannot be //instantiated
// Exists only to defeat instantiation.
private constructor() {
    fun newMessage(): String {
        return "This is a Singleton"
    }

    companion object {
        private val newInstance: Singleton? = null;

        //create an object of SingleObject
        var Instance: Singleton? = null

        init {
            val newInstance = Singleton()
            Instance = Singleton.newInstance
        }
    }
}